e3sm_diags/plot/arm_diags_plot.py

```diff
@@ -0,0 +1,647 @@
+    fig = plt.figure()  # Create figure
+
+    ax1 = fig.add_axes([0.15, 0.1, 0.8, 0.8])  # Create axes
+    xax = np.arange(1, 13, 1)
+
+    refs = vars_to_data.refs
+    test = vars_to_data.test
+    ax1.plot(
+        xax, test, "k", linewidth=2, label="Test: " + parameter.test_name_yrs
+    )  # +' ({0:.1f})'.format(np.mean(test.asma())))
+    test_season = get_seasonal_mean(test)
+    for i_ref, ref in enumerate(refs):
+        ref_season = get_seasonal_mean(ref)
+        ax1.plot(
+            xax,
+            ref,
+            line_color[i_ref],
+            linewidth=2,
+            label="Ref: " + parameter.ref_name,
+        )  # +' ({0:.1f})'.format(np.mean(ref.asma())))
+    my_xticks = ["J", "F", "M", "A", "M", "J", "J", "A", "S", "O", "N", "D"]
+    plt.xticks(xax, my_xticks)
+    plt.xlim(1, 12)
+    ymin, ymax = plt.gca().get_ylim()
+    plt.ylim(0.8 * ymin, 1.2 * ymax)
+    #     plt.ylim(ylim[va_ind])
+    plt.xlabel("Month")
+    plt.legend(loc="best", prop={"size": 10})
+
+    if var == "PRECT":
+        plt.ylabel("Total Precipitation Rate" + " (" + parameter.var_units + ")")
+    else:
+        plt.ylabel(parameter.var_name + " (" + parameter.var_units + ")")
+
+    # Add a table at the bottom of the axes
+    bias = test_season - ref_season
+    cell_text = np.round(np.vstack((test_season, ref_season, bias)), 2)
+    collabel = ("ANN", "DJF", "MAM", "JJA", "SON")
+    rows = ("Test", "Ref", "Bias")
+    plt.table(
+        cellText=cell_text,
+        rowLabels=rows,
+        colLabels=collabel,
+        alpha=0.8,
+        bbox=[0.15, 0.0, 0.8, 0.2],
+    )
+
+    # Save the figure.
+    output_file_name = parameter.output_file
+    plt.title(output_file_name.replace("-", " "))
+
+    for f in parameter.output_format:
+        f = f.lower().split(".")[-1]
+        fnm = os.path.join(
+            get_output_dir(parameter.current_set, parameter), output_file_name + "." + f
+        )
+        plt.savefig(fnm)
+        # Get the filename that the user has passed in and display that.
+        fnm = os.path.join(
+            get_output_dir(parameter.current_set, parameter),
+            output_file_name + "." + f,
+        )
+        logger.info(f"Plot saved in: {fnm}")
+
+    plt.close()
+
+
+def plot_diurnal_cycle(var, vars_to_data, parameter):
+    test = vars_to_data.test[0]
+    ref = vars_to_data.refs[0][0]
+    lst = vars_to_data.misc[0]
+    t_conv = lst[0][0]
+
+    output_file_name = parameter.output_file + "-" + "diurnal-cycle"
+
+    fig = plt.figure()  # Create figure
+    ax = fig.add_axes([0.15, 0.1, 0.8, 0.8])  # Create axes
+
+    for index in range(2):
+        if index == 0:
+            data = test
+            line_c = "k"
+            data_name = parameter.test_name_yrs
+        else:
+            data = ref
+            line_c = "r"
+            data_name = parameter.ref_name
+
+        time_freq = len(data)
+        res = int(24 / time_freq)
+        c, maxvalue, tmax = fastAllGridFT(data, [0])
```

Comment:
Accordingly, to use:
_fft_all_grid to replace fastAllGridFT

Reply:
Done